File: 1137 - N-th Tribonacci Number.js
/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function (n) {
    if (n == 0) {
        return 0;
    } else if (n <= 2) {
        return 1;
    }
    let a = 0, b = 1, c = 1, d = 0;
    for (let i = 0; i < n - 2; i++) {
        d = a + b + c;
        a = b;
        b = c;
        c = d;
    }
    return d;
};

File: 1137 - N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0
        elif n <= 2:
            return 1
        a, b, c = 0, 1, 1
        d = 0
        for i in range(n - 2):
            d = a + b + c
            a, b, c = b, c, d
        return d